### ISSUE TITLE
pattern for cell names

### ISSUE THREAD

**Issue by ay4me** [julian-upc/2015-algorithms#17]
How exactly is the pattern to name the cells in the matrix.
I thought it was something like Rowname + Columnname, where the Rowname is like I[4] if this is the forth row added for the pentomino I, but that seems to not fit the pattern used in the tests.

**Comment by ay4me** [comment]
So actually the pattern is right, it was a code error, so for the record MatrixCells are named as described above.

**Comment by ay4me** [comment]
Ok, I just noticed that if under D there are already two ones and we append a row with D as "key" then the new cell is named "D[0]" so this number does not correspond to the size (i.e. the numbers of 1s in each column). I don't see why we would want that.

Also the names of the cells are not unique, which we might want to have, depending on how we want to use them.

**Comment by neosonea** [comment]
I got to the same problem. construct_running_example fails due to that, right?
Intuitively, I thought of the same naming system and cannot think of another elaborate one.

But maybe the problem is where the rows are inserted? I always insert them as last row.

Or maybe it should be self.indexOfPiecePlacement[name]. I still do not get what this was meant for. And when should that be incremented, if not simultaneously with the columnsize?

**Comment by ay4me** [comment]
I do not have a problem with the tests anymore, because I changed the namings according to how the tests want them, but it took me a while to understand it.
If the third row is added to the D-column, its the first time, that D is the key for it, so the first entry in the row, that's why it gets D[0] instead of D[2] which would correspond to the current size of this column.
I inserted an extra counter for that. But I also fail to see why it's necessary, because I do not know how we want to use the cell names later on. I thought of just bringing it up in discussion tomorrow.
Hope that helped somehow.
